sys/net/gnrc/network_layer/ndp/internal/gnrc_ndp_internal.c

```diff
@@ -363,11 +363,32 @@ static bool _pio_from_iface_addr(gnrc_pktsnip_t **res, gnrc_ipv6_netif_addr_t *a
     return false;
 }
 
+static inline bool _check_prefixes(gnrc_ipv6_netif_addr_t *a, gnrc_ipv6_netif_addr_t *b)
+{
+    if ((a->prefix_len < 128U) && (a->prefix_len == b->prefix_len) &&
```

Comment:
this check is necessary to **not check** prefixes with a length of 128. Otherwise, there would be only one PIO for all `/128` prefixes, regardless of what the prefix is. I can change it to `a->prefix_len != 128` if desired